iwyu_include_picker.cc

```diff
@@ -1424,17 +1426,17 @@ void IncludePicker::ExpandRegexes() {
     for (const string& regex_key : filepath_include_map_regex_keys) {
       const vector<MappedInclude>& map_to = filepath_include_map_[regex_key];
       // Enclose the regex in ^(...)$ for full match.
-      std::regex regex(std::string("^(" + regex_key.substr(1) + ")$"));
-      if (std::regex_match(hdr, regex) &&
+      std::string regex("^(" + regex_key.substr(1) + ")$");
+      if (RegexMatch(regex_dialect, hdr, regex) &&
           !ContainsQuotedInclude(map_to, hdr)) {
```

Comment:
Assuming the regular expression is slower we should switch the expressions.

And if we do that we could inline `regex` so it is only constructed if actually needed.

Reply:
I have a followup planned to move the regex construction into RegexMatch itself, but I didn't want to complicate this PR with even more details.

By "switch the expressions", did you mean the map lookup and the RegexMatch call?

Reply:
> By "switch the expressions", did you mean the map lookup and the RegexMatch call?

Yes.

Reply:
👍 I'll look into it tomorrow. Possibly as a separate patch set. Thanks!

Reply:
> Assuming the regular expression is slower we should switch the expressions.

I tried both, and could not see any consistent difference. At first it looked like `llvm::Regex` was faster than the `Contains...` lookup and `std::regex` was _slower_ than the `Contains...` lookup, forming a nice dilemma :)

But with some more test repetitions it was mostly a draw, so I'll leave it as-is.

Added two other optimizations/fixups on top, though. Please take another look.

Reply:
> But with some more test repetitions it was mostly a draw, so I'll leave it as-is.

Exactly what I expected.

> Added two other optimizations/fixups on top, though. Please take another look.

Nice - I didn't look at those since I don't know what the code actually does. I also assumed they are "correct".